src/modules/anon_chat/index.js

```diff
@@ -0,0 +1,67 @@
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+const twitch = require('../../utils/twitch');
+const Raven = require('raven-js');
+
+const forcedURL = window.location.search && window.location.search.indexOf('bttv_anon_chat=true') > -1;
+
+class AnonChatModule {
+    constructor() {
+        settings.add({
+            id: 'anonChat',
+            name: 'Anon Chat',
+            defaultValue: false,
+            description: 'Join channels without appearing in chat'
+        });
+
+        this.ignoreNextDC = false;
+        watcher.on('load.chat', () => this.load());
+        watcher.on('chat.message', ($el, emberView) => this.onMessage($el, emberView));
+        settings.on('changed.anonChat', () => this.load());
+    }
+
+    load(force) {
```

Comment:
i'd much prefer an actual api for this rather than use load to toggle.. should be a .join and .part .. not hard since you can now easily store state outside the class